tests/test_library.py

```diff
@@ -0,0 +1,659 @@
+    tlp, backend = tlp
+    session = backend._session
+    session.mock_add_spec(("tidal_search", "artist", "artist.get_albums"))
+
+    artist = mocker.Mock()
+    artist.uri = "tidal:artist:1"
+
+    tidal_search = mocker.Mock()
+    tidal_search.return_value = ([artist], [], [])
+    session.artist.return_value = None
+    mocker.patch("mopidy_tidal.search.tidal_search", tidal_search)
+    assert not tlp.get_distinct(field, query={"any": "any"})
+    tidal_search.assert_called_once_with(session, query={"any": "any"}, exact=True)
+    session.artist.assert_called_once_with("1")
+
+
+def test_browse_wrong_uri(tlp):
+    tlp, backend = tlp
+    assert not tlp.browse("")
+    assert not tlp.browse("spotify:something:something_else")
+    assert not tlp.browse("tidal:album:oneid:oneidtoomany")
+
+
+def test_browse_root(tlp):
+    tlp, backend = tlp
+    assert tlp.browse("tidal:directory") == [
+        Ref(name="Genres", type="directory", uri="tidal:genres"),
+        Ref(name="Moods", type="directory", uri="tidal:moods"),
+        Ref(name="Mixes", type="directory", uri="tidal:mixes"),
+        Ref(name="My Artists", type="directory", uri="tidal:my_artists"),
+        Ref(name="My Albums", type="directory", uri="tidal:my_albums"),
+        Ref(name="My Playlists", type="directory", uri="tidal:my_playlists"),
+        Ref(name="My Tracks", type="directory", uri="tidal:my_tracks"),
+    ]
+
+
+def test_browse_artists(tlp, mocker, tidal_artists):
+    tlp, backend = tlp
+    session = backend._session
+    session.user.favorites.artists = tidal_artists
+    mocker.patch("mopidy_tidal.library.get_items", lambda x: x)
+    assert tlp.browse("tidal:my_artists") == [
+        Ref(name="Artist-0", type="artist", uri="tidal:artist:0"),
+        Ref(name="Artist-1", type="artist", uri="tidal:artist:1"),
+    ]
+
+
+def test_browse_albums(tlp, mocker, tidal_albums):
+    tlp, backend = tlp
+    session = backend._session
+    session.user.favorites.albums = tidal_albums
+    mocker.patch("mopidy_tidal.library.get_items", lambda x: x)
+    assert tlp.browse("tidal:my_albums") == [
+        Ref(name="Album-0", type="album", uri="tidal:album:0"),
+        Ref(name="Album-1", type="album", uri="tidal:album:1"),
+    ]
+
+
+def test_browse_tracks(tlp, mocker, tidal_tracks):
+    tlp, backend = tlp
+    session = backend._session
+    session.user.favorites.tracks = tidal_tracks
+    mocker.patch("mopidy_tidal.library.get_items", lambda x: x)
+    assert tlp.browse("tidal:my_tracks") == [
+        Ref(name="Track-0", type="track", uri="tidal:track:0:0:0"),
+        Ref(name="Track-1", type="track", uri="tidal:track:1:1:1"),
+    ]
+
+
+def test_browse_playlists(tlp, mocker):
+    tlp, backend = tlp
+    as_list = mocker.Mock()
+    uniq = object()
+    as_list.return_value = uniq
+    backend.playlists.as_list = as_list
+    assert tlp.browse("tidal:my_playlists") is uniq
+    as_list.assert_called_once_with()
+
+
+def test_moods_old_api(tlp, mocker):
```

Comment:
Perhaps we should consider if we want to keep support for the old tidalapi version in the tests.

Reply:
I think we should, as long as we keep support for it in the code.  100% line and branch coverage is *nice*, and whilst it doesn't find all bugs, at least it catches a fair few.

That said I think in e.g. a year we should drop support for the old tidalapi version everywhere.  This will tidy the logic up a lot, and a lot of duplicate tests can go as well.